src/audio/module_adapter/CMakeLists.txt

```diff
@@ -6,18 +6,18 @@ add_local_sources(sof module_adapter.c module/generic.c)
 if((NOT CONFIG_LIBRARY) OR CONFIG_LIBRARY_STATIC)
 	if(CONFIG_COMP_VOLUME)
 	add_local_sources(sof
-		module/volume/volume_generic.c
-		module/volume/volume_hifi3.c
-		module/volume/volume_hifi4.c
-		module/volume/volume_generic_with_peakvol.c
-		module/volume/volume_hifi3_with_peakvol.c
-		module/volume/volume_hifi4_with_peakvol.c
-		module/volume/volume_common.c)
+		../volume/volume_generic.c
+		../volume/volume_hifi3.c
+		../volume/volume_hifi4.c
+		../volume/volume_generic_with_peakvol.c
+		../volume/volume_hifi3_with_peakvol.c
+		../volume/volume_hifi4_with_peakvol.c
+		../volume/volume_common.c)
```

Comment:
This need to be in a Cmakefile in the module directory i.e. 
```src/audio/volume/CMakeList.txt```

Reply:
also no need to use ```_with_``` in file names, I would just call the files (using above order)
```
volume/generic.c
volume/hifi3.c
volume/hifi4.c
volume/peakvol.c
volume/hifi3_peakvol.c
volume/hifi4_peakvol.c
volume/common.c
```

Reply:
@lgirdwood @btian1  both are fine to me, but we have still kept the module/component prefix in filenames (e.g. we have pipeline/pipeline-foobar.c so to be consistent, we should have volume/volume_generic.c:

```
├── src
│   ├── CMakeLists.txt
│   ├── src.c
│   ├── src_generic.c
│   ├── src_hifi2ep.c
│   ├── src_hifi3.c
│   └── src_hifi4.c
├── tdfb
│   ├── CMakeLists.txt
│   ├── tdfb.c
│   ├── tdfb_direction.c
│   ├── tdfb_generic.c
│   ├── tdfb_hifi3.c
│   └── tdfb_hifiep.c
....
```

Personally I'd rather have a module prefix in filename. It's a bit confusing to have "generic.c" open in editor and not immediately see what instance of generic.c this one is (maybe just an old grumpy emacs user here). But oh well, just my personal preference.

Reply:
@kv2019i btw, what does mtrace show wrt filenames ?

Reply:
@kv2019i logging uses module names and does not use the filename (e.g. we init with "LOG_MODULE_REGISTER(volume, CONFIG_SOF_LOG_LEVEL);").

Reply:
ok, I'm happy that logging shows the correct detail and ```volume_``` prefix for file names to help some editors.

Reply:
> Personally I'd rather have a module prefix in filename. It's a bit confusing to have "generic.c" open in editor and not immediately see what instance of generic.c this one is

A weak +1 on this. In general I can live with either solution, that's why I didn't comment on this. And (as many know) I'm all against redundancy so - if the folder is already called "volume" why should files in it have a "volume_" prefix? Same for function and type names: should static functions use a "volume_" prefix or not? But since some time my preference shifted towards this redundancy - to use a prefix in all those cases for pretty much the same reason - for editor, shell and tag use. Imagine if all drivers and modules had functions called `init()`, `configure()`, `copy()`, etc.

Reply:
> if the folder is already called "volume" why should files in it have a "volume_" prefix?

Because most editors suck? :-) They don't display the end of the path.

It also depends on the basename, for instance we have multiple `interrupt.c` but at least "interrupt" means _something_ whereas "generic" and "common" carry _zero_ information. So the latter needs a prefix more than the former.